portality/util.py

```diff
@@ -125,7 +125,18 @@ def normalise_issn(issn):
             issn = ("0" * (8 - len(issn))) + issn
             return issn[:4] + "-" + issn[4:]
 
+
 def load_file(filename):
     with open(filename, 'rb') as f:
         content = f.read()
-    return content
\ No newline at end of file
+    return content
+
+
+def unicode_dict(d):
+    """ Recursively convert dictionary keys to unicode """
+    if isinstance(d, dict):
+        return dict((unicode(k), unicode_dict(v)) for k, v in d.items())
```

Comment:
there are dictionary comprehensions in Python nowadays as well, but I presume this is already-tested code so best stick to it :)